### PR TITLE
Add Tableau Config to Invite (Backend)

### PR DESCRIPTION
**Pull request by Jtang-1**
## Product Description

no user facing changes

## Technical Summary

[Jira Ticket](https://dimagi.atlassian.net/browse/USH-4451)
Related to model change [PR](https://github.com/dimagi/commcare-hq/pull/34445)
This PR implements the backend changes needed to configure a web user with a "Tableau Role" and "Tableau Groups" via web user invite workflow. 

## Feature Flag

[Tableau User Syncing](https://www.commcarehq.org/hq/flags/edit/tableau_user_syncing/)

### Safety story

I locally tested that 
- when invitations contain values in both "tableau_role" and "tableau_groups," they successfully create or update the associated TableauUser object for the respective web user, ensuring that the TableauUser possesses these specified properties.
- If the web user already exists within domain2 (along with its corresponding TableauUser object), accepting an invitation to domain1 will result in the update of the Tableau Role and Tableau Groups for that web user across both domain1 and domain2.

### Automated test coverage

I added test:
-  That the new tableau util `get_tableau_groups_by_ids`  returns the correct tableau group names
-  That `add_web_user` calls correctly calls the functions needed to add the tableau role and groups to the web user

### QA Plan

no QA. QA will be done when the UI portion is complete for full feature testing.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by AddisonDunn — APPROVED** [review]
Muy bien

If you haven't already, you might play around with adding a user to a 2nd domain where tableau user syncing is enabled and the domain is connected to the same Tableau site as the first. Or you might leave that for QA. I don't see anything here that suggests that config wouldn't work but just an FYI it'd be good to test at some point with this feature.

File: corehq/apps/users/models.py [hunk 1]
```diff
@@ -530,9 +530,12 @@ def add_domain_membership(self, domain, timezone=None, **kwargs):
         self.domains.append(domain)
 
     def add_as_web_user(self, domain, role, primary_location_id=None,
-                        assigned_location_ids=None, program_id=None, profile=None):
+                        assigned_location_ids=None, program_id=None, profile=None,
+                        tableau_role=None, tableau_group_ids=None):
         if assigned_location_ids is None:
             assigned_location_ids = []
+        if tableau_group_ids is None:
+            tableau_group_ids = []
```
[the remark this answers is not in the document]
Comment (Jtang-1) [reply corehq/apps/users/models.py]: [ab14cb3](https://github.com/dimagi/commcare-hq/pull/34606/commits/ab14cb36f78f678452999f36281cbea61515a089)

File: corehq/apps/users/tests/test_user_model.py [hunk 2]
```diff
@@ -178,20 +178,43 @@ def test_update_app_metadata(self):
...
+        user = self.user_class.create(
+            domain=domain,
+            username=username,
+            password='***',
+            created_by=None,
+            created_via=None,
+            role_id=role_id
+        )
+        self.addCleanup(user.delete, None, None)
+        return user
```
[the remark this answers is not in the document]
Comment (Jtang-1) [reply corehq/apps/users/tests/test_user_model.py:209]: [d5e7de7](https://github.com/dimagi/commcare-hq/pull/34606/commits/d5e7de72888da30c3731e162e0fdf02b5460c1b5)

**Comment by Jtang-1** [comment]
Thanks for the review Addison, I tested the scenario you described with two different domains and tableau configs are updated as expected between domains upon Invitation acceptance. 

Also updated the description block for the local testing done per your comment

**Review by AddisonDunn — APPROVED** [review]

**Pull request merged**